src/cascadia/TerminalConnection/AzureConnection.cpp

```diff
@@ -0,0 +1,885 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "AzureConnection.h"
+#include "AzureClientID.h"
+#include "AzureConnectionStrings.h"
+#include <sstream>
+#include <stdlib.h>
+
+#include "AzureConnection.g.cpp"
+
+#include "../../types/inc/Utils.hpp"
+
+using namespace ::Microsoft::Console;
+
+using namespace utility;
+using namespace web;
+using namespace web::json;
+using namespace web::http;
+using namespace web::http::client;
+using namespace web::websockets::client;
+using namespace concurrency::streams;
+using namespace winrt::Windows::Security::Credentials;
+
+namespace winrt::Microsoft::Terminal::TerminalConnection::implementation
+{
+    AzureConnection::AzureConnection(const uint32_t initialRows, const uint32_t initialCols) :
+        _initialRows{ initialRows },
+        _initialCols{ initialCols }
+    {
+    }
+
+    // Method description:
+    // - ascribes to the ITerminalConnection interface
+    // - registers an output event handler
+    // Arguments:
+    // - the handler
+    // Return value:
+    // - the event token for the handler
+    winrt::event_token AzureConnection::TerminalOutput(Microsoft::Terminal::TerminalConnection::TerminalOutputEventArgs const& handler)

```

Comment:
I think we might have some `DEFINE_EVENT_HANDLER` macros lying around so you don't need to put these in here manually, but they're fine for now.

Reply:
Ah I was just following the way ConhostConnection.cpp did it